ports/nrf/common-hal/_bleio/Adapter.c

```diff
@@ -479,15 +479,15 @@ mp_obj_t common_hal_bleio_adapter_start_scan(bleio_adapter_obj_t *self, uint8_t*
         mp_raise_ValueError(translate("non-zero timeout must be > 0.01"));
     }
 
-    if (nrf_timeout) {
+    if (nrf_timeout == 0) {
```

Comment:
Does 0.5f round down? Does setting unlimited still work?

Reply:
Yes, for `timeout=0.0f`, this reduces to `uint32_t nrf_timeout = 0.0f + 0.5f;`, and casting a `float` to an integer in C truncates, so `nrf_timeout` becomes `0`.

To triple-check, I actually tried `timeout=0.0`, and it scanned forever (well, I didn't wait that long).